specifyweb/frontend/js_src/lib/components/QueryBuilder/CheckReadAccess.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { commonText } from '../../localization/common';
+import { queryText } from '../../localization/query';
+import { RA } from '../../utils/types';
+import { Ul } from '../Atoms';
+import { SerializedResource } from '../DataModel/helperTypes';
+import { SpQuery, Tables } from '../DataModel/types';
+import { Dialog } from '../Molecules/Dialog';
+import { TableIcon } from '../Molecules/TableIcon';
+import { hasTablePermission } from '../Permissions/helpers';
+import { QueryFieldSpec } from './fieldSpec';
+
+export function CheckReadAccess({
+  query,
+}: {
+  readonly query: SerializedResource<SpQuery>;
+}): JSX.Element | null {
+  const [noAccessTables, setNoAccessTables] = React.useState<RA<keyof Tables>>(
+    []
+  );
+
+  function accessTables(query: SerializedResource<SpQuery>): RA<keyof Tables> {
```

Comment:
Bad function name. It should either return the tables, in which case it should be called `getNoAccessTables`, or modify the tables, in which case it could be called `findNoAccessTables`